### PR TITLE
feat: error handling and array expansion

### LINKED ISSUE #2425: [Feature] Supplement Math Formulas for Formula Engine
### Initial checklist

- [X] Is this really a problem?
- [X] I have searched the Github Issues for similar issues, but did not find anything.

How to contribute formulas: https://univer.ai/guides/sheet/customization/formula#how-to-add-formulas-in-formula-engine

It is highly recommended that you refer to the algorithms of similar formulas that have been implemented

https://github.com/dream-num/univer/blob/dev/packages/engine-formula/src/functions/math/function-map.ts

Especially for

ASIN
ASINH
ATAN
ATAN2
ATANH
COS

These, like Acos, already have corresponding underlying implementations in ArrayValueObject

### Solution

Formulas List:
- [ ] ACOTH
- [ ] ASIN
- [ ] ASINH
- [ ] ATAN
- [ ] ATAN2
- [ ] ATANH
- [ ] COS
- [ ] COSH
- [ ] COT
- [ ] COTH
- [ ] CSC
- [ ] CSCH

### PR DESCRIPTION
**Pull request by abdelrahman830a**
close #2425

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/engine-formula/src/functions/math/asin/index.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+function asin(num: BaseValueObject) {
+    let currentValue = num.getValue();
+
+    if (num.isBoolean()) {
+        currentValue = currentValue ? 1 : 0;
+    }
+
+    if (!Number.isFinite(currentValue) || Number(currentValue) < -1 || Number(currentValue) > 1) {
+        return ErrorValueObject.create(ErrorType.VALUE);
```
Comment (Dushusir) [inline packages/engine-formula/src/functions/math/asin/index.ts:58]: should reports #NUM!
image

File: packages/engine-formula/src/functions/math/atan2/index.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    if (num2.isBoolean()) {
+        currentValue2 = currentValue2 ? 1 : 0;
+    }
+
+    if (!Number.isFinite(currentValue1) || !Number.isFinite(currentValue2)) {
+        return ErrorValueObject.create(ErrorType.VALUE);
+    }
+
+    const result = Math.atan2(Number(currentValue1), Number(currentValue2));
```
Comment (Dushusir) [inline packages/engine-formula/src/functions/math/atan2/index.ts:65]: The two parameters of this formula may be arrays of different sizes. In this case, you need to use expandArrayValueObject to expand the two input parameters. Refer to the usage in [Text](https://github.com/dream-num/univer/blob/dev/packages/engine-formula/src/functions/text/text/index.ts).

However, I suggest using the existing method [atan2](https://github.com/dream-num/univer/blob/cfdc48faa3edda7d4131f1b1cdc746420a6b836d/packages/engine-formula/src/engine/value-object/array-value-object.ts#L1129) in ArrayValueObejct. If it doesn't work, expand it yourself

**Review by Dushusir — CHANGES_REQUESTED** [review]

**Comment by Dushusir** [comment]
Please check the PR title and test errors, and make sure it passes all CI (except Univer Merge Assistant, which is manually triggered by our QA)

File: packages/engine-formula/src/functions/math/acos/index.ts [hunk 3]
```diff
@@ -1,38 +1,68 @@
...
+
+function calculateAcos(variant: BaseValueObject) {
+    let value = variant.getValue();
+
+    if (variant.isBoolean()) {
+        value = value ? 1 : 0;
+    }
+
+    if (typeof value !== 'number' || value < -1 || value > 1) {
+        return ErrorValueObject.create(ErrorType.VALUE);
```
Comment (Dushusir) [inline packages/engine-formula/src/functions/math/acos/index.ts:58]: =ACOS(3) reports #NUM!
image

**Comment by Dushusir** [comment]
Please refer to the second step to complete the internationalization content, refer to https://univer.ai/guides/sheet/customization/formula#how-to-implement-the-formula, just like adding https://github.com/dream-num/univer/blob/56c6e1269791dfb835f1af7ccdc381518f97e6b5/packages/sheets-formula/src/locale/function-list/math/en-US.ts#L115

When you enter the formula, you can trigger the auto-completion, which means that the internationalization content has been added successfully.
image

File: packages/engine-formula/src/functions/math/acos/index.ts (hunk 3 above)
Comment (abdelrahman830a) [reply packages/engine-formula/src/functions/math/acos/index.ts:58]: the Math.acos() function in JavaScript is defined only for values in the range -1 through 1. If the input is outside this range, Math.acos() returns NaN.

**Pull request closed without merging**